File: binary-search/binary-search-2.c
int search(int* nums, int numsSize, int target){
    int left = 0, right = numsSize;
    while (left < right) {
        int mid = (left + right) / 2;
        if (nums[mid] == target) {
            return mid;
        } else if (nums[mid] < target && left != mid) {
            left = mid;
        } else if (nums[mid] > target && right != mid) {
            right = mid;
        } else {
            break;
        }
    }
    return -1;
}

File: binary-search/binary-search-2.java
class Solution {
    public int search(int[] nums, int target) {
        if (nums.length < 1) {
            return -1;
        }

        int lo = 0, hi = nums.length - 1;
        while (lo < hi) {
            int mid = lo + (hi - lo) / 2;
            if (nums[mid] < target) {
                lo = mid + 1;
            } else if (nums[mid] > target) {
                hi = mid - 1;
            } else {
                return mid;
            }
        }

        if (hi >= 0 && nums[hi] == target) {
            return hi;
        }

        return -1;
    }
}